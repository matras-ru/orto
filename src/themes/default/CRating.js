export const stateDefault = '#dedede';
export const stateActive = '#fda368';

export const starsWrapper = 'inline-block';
export const labelWrapper = 'inline-block align-middle';

export const stateCursorPointer = 'cursor-pointer';

export default {
    stateDefault,
    stateActive,

    starsWrapper,
    labelWrapper,

    stateCursorPointer
};
