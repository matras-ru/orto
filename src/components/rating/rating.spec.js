/* eslint-env jest */
import { createLocalVue, mount } from '@vue/test-utils';
import CRating from './rating';

describe('Rating basic', () => {
    it('is rendered', () => {
        const wrapper = mount(CRating, {
            propsData: {
                id: 'test1'
            }
        });

        const stars = wrapper.find('span svg');

        expect(wrapper.element.tagName).toEqual('DIV');
        expect(stars.element.tagName).toEqual('svg');
        expect(wrapper.html()).toMatchSnapshot();

        wrapper.destroy();
    });
});

describe('Rating with label', () => {
    it('is rendered', () => {
        const scale = [
            { id: 1, label: 'Ужасно', value: 1 },
            { id: 2, label: 'Плохо', value: 2 },
            { id: 3, label: 'Нормально', value: 3 },
            { id: 4, label: 'Хорошо', value: 4 },
            { id: 5, label: 'Отлично', value: 5 }
        ];

        const wrapper = mount(CRating, {
            propsData: {
                scale,
                id: 'test1'
            }
        });

        const stars = wrapper.find('span svg');
        const label = wrapper.find('div');

        expect(wrapper.element.tagName).toEqual('DIV');
        expect(stars.element.tagName).toEqual('svg');
        expect(label.element.tagName).toEqual('DIV');
        expect(wrapper.vm.scale).toBe(scale);
        expect(wrapper.html()).toMatchSnapshot();

        wrapper.destroy();
    });
});

describe('Rating with rating number', () => {
    it('is rendered', () => {
        const wrapper = mount(CRating, {
            propsData: {
                rating: '4.3',
                id: 'test1'
            }
        });

        const stars = wrapper.find('span svg');
        const rating = wrapper.find('div');

        expect(wrapper.element.tagName).toEqual('DIV');
        expect(stars.element.tagName).toEqual('svg');
        expect(rating.element.tagName).toEqual('DIV');
        expect(wrapper.vm.rating).toBe('4.3');
        expect(wrapper.html()).toMatchSnapshot();

        wrapper.destroy();
    });
});

describe('Rating events', () => {
    it('should emit star selected event when selected', () => {
        const wrapper = mount(CRating);

        const stars = wrapper.find('span svg');
        stars.trigger('click');
        expect(stars.emitted()).toHaveProperty('starSelected');

        wrapper.destroy();
    });

    it('should emit star mouse move event when mouse move', () => {
        const wrapper = mount(CRating);

        const stars = wrapper.find('span svg');
        stars.trigger('mousemove');
        expect(stars.emitted()).toHaveProperty('starMouseMove');

        wrapper.destroy();
    });
});

describe('Rating readonly and events', () => {
    it('is rendered', () => {
        const wrapper = mount(CRating, {
            propsData: {
                id: 'test1',
                readOnly: true,
                rating: '4.5'
            }
        });

        const stars = wrapper.find('span svg');

        expect(wrapper.element.tagName).toEqual('DIV');
        expect(stars.element.tagName).toEqual('svg');
        expect(wrapper.vm.readOnly).toBe(true);
        expect(wrapper.vm.rating).toBe('4.5');
        expect(wrapper.html()).toMatchSnapshot();

        wrapper.destroy();
    });

    it('should emit star selected event when selected', () => {
        const wrapper = mount(CRating, {
            propsData: {
                readOnly: true,
                rating: '4.5'
            }
        });

        const stars = wrapper.find('span svg');
        stars.trigger('click');
        expect(wrapper.emitted()).not.toHaveProperty('starSelected');

        wrapper.destroy();
    });

    it('should emit star mouse move event when mouse move', async () => {
        const wrapper = mount(CRating, {
            propsData: {
                readOnly: true,
                rating: '4.5'
            }
        });

        const stars = wrapper.find('span svg');
        await stars.trigger('mousemove');
        expect(wrapper.emitted()).not.toHaveProperty('starMouseMove');

        wrapper.destroy();
    });
});

describe('Rating v-model', () => {
    it('v-model', () => {
        const localVue = new createLocalVue();

        const App = localVue.extend({
            data() {
                return {
                    model: undefined
                };
            },

            render(h) {
                return h(CRating, {
                    props: {
                        id: 'test1',
                        gradeModel: this.model
                    },
                    on: {
                        'star-selected': val => {
                            this.model = val;
                        }
                    }
                });
            }
        });

        const wrapper = mount(App, {
            localVue: localVue
        });

        expect(wrapper.vm.model).toEqual(undefined);
        wrapper.setData({ model: 5 });
        expect(wrapper.vm.model).toEqual(5);
        expect(wrapper.html()).toMatchSnapshot();

        wrapper.destroy();
    });
});
