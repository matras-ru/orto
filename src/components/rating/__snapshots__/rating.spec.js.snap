// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Rating basic is rendered 1`] = `<div><span class="inline-block stateCursorPointer"><svg viewBox="0 0 50 50" xmlns="http://www.w3.org/2000/svg" width="2rem" class="inline-block"><linearGradient id="test10" x1="0" x2="100%" y1="0" y2="0"><stop offset="0%" stop-color="#dedede"></stop><stop offset="0%" stop-color="#dedede"></stop></linearGradient><polygon points="19.8,2.2,6.6,43.56,39.6,17.16,0,17.16,33,43.56" fill="url(#test10)"></polygon></svg><svg viewBox="0 0 50 50" xmlns="http://www.w3.org/2000/svg" width="2rem" class="inline-block"><linearGradient id="test11" x1="0" x2="100%" y1="0" y2="0"><stop offset="0%" stop-color="#dedede"></stop><stop offset="0%" stop-color="#dedede"></stop></linearGradient><polygon points="19.8,2.2,6.6,43.56,39.6,17.16,0,17.16,33,43.56" fill="url(#test11)"></polygon></svg><svg viewBox="0 0 50 50" xmlns="http://www.w3.org/2000/svg" width="2rem" class="inline-block"><linearGradient id="test12" x1="0" x2="100%" y1="0" y2="0"><stop offset="0%" stop-color="#dedede"></stop><stop offset="0%" stop-color="#dedede"></stop></linearGradient><polygon points="19.8,2.2,6.6,43.56,39.6,17.16,0,17.16,33,43.56" fill="url(#test12)"></polygon></svg><svg viewBox="0 0 50 50" xmlns="http://www.w3.org/2000/svg" width="2rem" class="inline-block"><linearGradient id="test13" x1="0" x2="100%" y1="0" y2="0"><stop offset="0%" stop-color="#dedede"></stop><stop offset="0%" stop-color="#dedede"></stop></linearGradient><polygon points="19.8,2.2,6.6,43.56,39.6,17.16,0,17.16,33,43.56" fill="url(#test13)"></polygon></svg><svg viewBox="0 0 50 50" xmlns="http://www.w3.org/2000/svg" width="2rem" class="inline-block"><linearGradient id="test14" x1="0" x2="100%" y1="0" y2="0"><stop offset="0%" stop-color="#dedede"></stop><stop offset="0%" stop-color="#dedede"></stop></linearGradient><polygon points="19.8,2.2,6.6,43.56,39.6,17.16,0,17.16,33,43.56" fill="url(#test14)"></polygon></svg></span></div>`;

exports[`Rating readonly and events is rendered 1`] = `
<div><span class="inline-block"><svg viewBox="0 0 50 50" xmlns="http://www.w3.org/2000/svg" width="2rem" class="inline-block"><linearGradient id="test10" x1="0" x2="100%" y1="0" y2="0"><stop offset="100%" stop-color="#fda368"></stop><stop offset="100%" stop-color="#dedede"></stop></linearGradient><polygon points="19.8,2.2,6.6,43.56,39.6,17.16,0,17.16,33,43.56" fill="url(#test10)"></polygon></svg><svg viewBox="0 0 50 50" xmlns="http://www.w3.org/2000/svg" width="2rem" class="inline-block"><linearGradient id="test11" x1="0" x2="100%" y1="0" y2="0"><stop offset="100%" stop-color="#fda368"></stop><stop offset="100%" stop-color="#dedede"></stop></linearGradient><polygon points="19.8,2.2,6.6,43.56,39.6,17.16,0,17.16,33,43.56" fill="url(#test11)"></polygon></svg><svg viewBox="0 0 50 50" xmlns="http://www.w3.org/2000/svg" width="2rem" class="inline-block"><linearGradient id="test12" x1="0" x2="100%" y1="0" y2="0"><stop offset="100%" stop-color="#fda368"></stop><stop offset="100%" stop-color="#dedede"></stop></linearGradient><polygon points="19.8,2.2,6.6,43.56,39.6,17.16,0,17.16,33,43.56" fill="url(#test12)"></polygon></svg><svg viewBox="0 0 50 50" xmlns="http://www.w3.org/2000/svg" width="2rem" class="inline-block"><linearGradient id="test13" x1="0" x2="100%" y1="0" y2="0"><stop offset="100%" stop-color="#fda368"></stop><stop offset="100%" stop-color="#dedede"></stop></linearGradient><polygon points="19.8,2.2,6.6,43.56,39.6,17.16,0,17.16,33,43.56" fill="url(#test13)"></polygon></svg><svg viewBox="0 0 50 50" xmlns="http://www.w3.org/2000/svg" width="2rem" class="inline-block"><linearGradient id="test14" x1="0" x2="100%" y1="0" y2="0"><stop offset="100%" stop-color="#fda368"></stop><stop offset="100%" stop-color="#dedede"></stop></linearGradient><polygon points="19.8,2.2,6.6,43.56,39.6,17.16,0,17.16,33,43.56" fill="url(#test14)"></polygon></svg></span>
  <div class="inline-block align-middle"> - 4.5</div>
</div>
`;

exports[`Rating v-model v-model 1`] = `<div><span class="inline-block stateCursorPointer"><svg viewBox="0 0 50 50" xmlns="http://www.w3.org/2000/svg" width="2rem" class="inline-block"><linearGradient id="test10" x1="0" x2="100%" y1="0" y2="0"><stop offset="0%" stop-color="#dedede"></stop><stop offset="0%" stop-color="#dedede"></stop></linearGradient><polygon points="19.8,2.2,6.6,43.56,39.6,17.16,0,17.16,33,43.56" fill="url(#test10)"></polygon></svg><svg viewBox="0 0 50 50" xmlns="http://www.w3.org/2000/svg" width="2rem" class="inline-block"><linearGradient id="test11" x1="0" x2="100%" y1="0" y2="0"><stop offset="0%" stop-color="#dedede"></stop><stop offset="0%" stop-color="#dedede"></stop></linearGradient><polygon points="19.8,2.2,6.6,43.56,39.6,17.16,0,17.16,33,43.56" fill="url(#test11)"></polygon></svg><svg viewBox="0 0 50 50" xmlns="http://www.w3.org/2000/svg" width="2rem" class="inline-block"><linearGradient id="test12" x1="0" x2="100%" y1="0" y2="0"><stop offset="0%" stop-color="#dedede"></stop><stop offset="0%" stop-color="#dedede"></stop></linearGradient><polygon points="19.8,2.2,6.6,43.56,39.6,17.16,0,17.16,33,43.56" fill="url(#test12)"></polygon></svg><svg viewBox="0 0 50 50" xmlns="http://www.w3.org/2000/svg" width="2rem" class="inline-block"><linearGradient id="test13" x1="0" x2="100%" y1="0" y2="0"><stop offset="0%" stop-color="#dedede"></stop><stop offset="0%" stop-color="#dedede"></stop></linearGradient><polygon points="19.8,2.2,6.6,43.56,39.6,17.16,0,17.16,33,43.56" fill="url(#test13)"></polygon></svg><svg viewBox="0 0 50 50" xmlns="http://www.w3.org/2000/svg" width="2rem" class="inline-block"><linearGradient id="test14" x1="0" x2="100%" y1="0" y2="0"><stop offset="0%" stop-color="#dedede"></stop><stop offset="0%" stop-color="#dedede"></stop></linearGradient><polygon points="19.8,2.2,6.6,43.56,39.6,17.16,0,17.16,33,43.56" fill="url(#test14)"></polygon></svg></span></div>`;

exports[`Rating with label is rendered 1`] = `<div><span class="inline-block stateCursorPointer"><svg viewBox="0 0 50 50" xmlns="http://www.w3.org/2000/svg" width="2rem" class="inline-block"><linearGradient id="test10" x1="0" x2="100%" y1="0" y2="0"><stop offset="0%" stop-color="#dedede"></stop><stop offset="0%" stop-color="#dedede"></stop></linearGradient><polygon points="19.8,2.2,6.6,43.56,39.6,17.16,0,17.16,33,43.56" fill="url(#test10)"></polygon></svg><svg viewBox="0 0 50 50" xmlns="http://www.w3.org/2000/svg" width="2rem" class="inline-block"><linearGradient id="test11" x1="0" x2="100%" y1="0" y2="0"><stop offset="0%" stop-color="#dedede"></stop><stop offset="0%" stop-color="#dedede"></stop></linearGradient><polygon points="19.8,2.2,6.6,43.56,39.6,17.16,0,17.16,33,43.56" fill="url(#test11)"></polygon></svg><svg viewBox="0 0 50 50" xmlns="http://www.w3.org/2000/svg" width="2rem" class="inline-block"><linearGradient id="test12" x1="0" x2="100%" y1="0" y2="0"><stop offset="0%" stop-color="#dedede"></stop><stop offset="0%" stop-color="#dedede"></stop></linearGradient><polygon points="19.8,2.2,6.6,43.56,39.6,17.16,0,17.16,33,43.56" fill="url(#test12)"></polygon></svg><svg viewBox="0 0 50 50" xmlns="http://www.w3.org/2000/svg" width="2rem" class="inline-block"><linearGradient id="test13" x1="0" x2="100%" y1="0" y2="0"><stop offset="0%" stop-color="#dedede"></stop><stop offset="0%" stop-color="#dedede"></stop></linearGradient><polygon points="19.8,2.2,6.6,43.56,39.6,17.16,0,17.16,33,43.56" fill="url(#test13)"></polygon></svg><svg viewBox="0 0 50 50" xmlns="http://www.w3.org/2000/svg" width="2rem" class="inline-block"><linearGradient id="test14" x1="0" x2="100%" y1="0" y2="0"><stop offset="0%" stop-color="#dedede"></stop><stop offset="0%" stop-color="#dedede"></stop></linearGradient><polygon points="19.8,2.2,6.6,43.56,39.6,17.16,0,17.16,33,43.56" fill="url(#test14)"></polygon></svg></span></div>`;

exports[`Rating with rating number is rendered 1`] = `
<div><span class="inline-block stateCursorPointer"><svg viewBox="0 0 50 50" xmlns="http://www.w3.org/2000/svg" width="2rem" class="inline-block"><linearGradient id="test10" x1="0" x2="100%" y1="0" y2="0"><stop offset="100%" stop-color="#fda368"></stop><stop offset="100%" stop-color="#dedede"></stop></linearGradient><polygon points="19.8,2.2,6.6,43.56,39.6,17.16,0,17.16,33,43.56" fill="url(#test10)"></polygon></svg><svg viewBox="0 0 50 50" xmlns="http://www.w3.org/2000/svg" width="2rem" class="inline-block"><linearGradient id="test11" x1="0" x2="100%" y1="0" y2="0"><stop offset="100%" stop-color="#fda368"></stop><stop offset="100%" stop-color="#dedede"></stop></linearGradient><polygon points="19.8,2.2,6.6,43.56,39.6,17.16,0,17.16,33,43.56" fill="url(#test11)"></polygon></svg><svg viewBox="0 0 50 50" xmlns="http://www.w3.org/2000/svg" width="2rem" class="inline-block"><linearGradient id="test12" x1="0" x2="100%" y1="0" y2="0"><stop offset="100%" stop-color="#fda368"></stop><stop offset="100%" stop-color="#dedede"></stop></linearGradient><polygon points="19.8,2.2,6.6,43.56,39.6,17.16,0,17.16,33,43.56" fill="url(#test12)"></polygon></svg><svg viewBox="0 0 50 50" xmlns="http://www.w3.org/2000/svg" width="2rem" class="inline-block"><linearGradient id="test13" x1="0" x2="100%" y1="0" y2="0"><stop offset="100%" stop-color="#fda368"></stop><stop offset="100%" stop-color="#dedede"></stop></linearGradient><polygon points="19.8,2.2,6.6,43.56,39.6,17.16,0,17.16,33,43.56" fill="url(#test13)"></polygon></svg><svg viewBox="0 0 50 50" xmlns="http://www.w3.org/2000/svg" width="2rem" class="inline-block"><linearGradient id="test14" x1="0" x2="100%" y1="0" y2="0"><stop offset="100%" stop-color="#fda368"></stop><stop offset="100%" stop-color="#dedede"></stop></linearGradient><polygon points="19.8,2.2,6.6,43.56,39.6,17.16,0,17.16,33,43.56" fill="url(#test14)"></polygon></svg></span>
  <div class="inline-block align-middle"> - 4.3</div>
</div>
`;
